plugin/src/main/java/org/opensearch/ml/action/deploy/TransportDeployModelAction.java

```diff
@@ -108,15 +107,50 @@ public TransportDeployModelAction(
         this.mlModelManager = mlModelManager;
         this.mlStats = mlStats;
         allowCustomDeploymentPlan = ML_COMMONS_ALLOW_CUSTOM_DEPLOYMENT_PLAN.get(settings);
+        filterByEnabled = ML_COMMONS_VALIDATE_BACKEND_ROLES.get(settings);
         clusterService
             .getClusterSettings()
             .addSettingsUpdateConsumer(ML_COMMONS_ALLOW_CUSTOM_DEPLOYMENT_PLAN, it -> allowCustomDeploymentPlan = it);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(ML_COMMONS_VALIDATE_BACKEND_ROLES, it -> filterByEnabled = it);
+
     }
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<MLDeployModelResponse> listener) {
         MLDeployModelRequest deployModelRequest = MLDeployModelRequest.fromActionRequest(request);
         String modelId = deployModelRequest.getModelId();
+        log.info(modelId);
+
+        MLModelGetRequest mlModelGetRequest = new MLModelGetRequest(modelId, false);
```

Comment:
Why create a new `MLModelGetRequest` ? I see the `mlModelGetRequest` only being used by line 125 to get `mlModelGetRequest.isReturnContent()`. As you have hard coded `returnContent` as false in this line (line 124), then the `mlModelGetRequest.isReturnContent()` will always return `false`, right?